src/psyclone/tests/psyir/frontend/fparser2_part_ref_test.py

```diff
@@ -0,0 +1,157 @@
+    # checks that the symbol is declared.
+    with pytest.raises(SymbolError) as error:
+        processor.process_nodes(fake_parent, [fparser2part_ref])
+    assert "No Symbol found for name 'x'." in str(error.value)
+
+
+@pytest.mark.usefixtures("f2008_parser")
+def test_handling_part_ref():
+    '''Test that fparser2 Part_Ref is converted to the expected PSyIR
+    tree structure.
+
+    '''
+    reader = FortranStringReader("x(2)=1")
+    fparser2part_ref = Execution_Part.match(reader)[0][0]
+
+    fake_parent = KernelSchedule('kernel')
+    fake_parent.symbol_table.add(DataSymbol('x', INTEGER_TYPE))
+
+    processor = Fparser2Reader()
+    processor.process_nodes(fake_parent, [fparser2part_ref])
+
+    assert len(fake_parent.children) == 1
+    assignment = fake_parent.children[0]
+    assert len(assignment.children) == 2
+    new_node = assignment.children[0]
+    assert isinstance(new_node, ArrayReference)
+    assert new_node.name == "x"
+    assert len(new_node.children) == 1  # Array dimensions
+
+
+@pytest.mark.usefixtures("f2008_parser")
+def test_handling_part_ref_expression():
+    '''Test that fparser2 Part_Ref is converted to the expected PSyIR
+    tree structure when there is a complex expression.
+
+    '''
+    # Parse a complex array expression
+    reader = FortranStringReader("x(i+3,j-4,(z*5)+1)=1")
+    fparser2part_ref = Execution_Part.match(reader)[0][0]
+
+    fake_parent = KernelSchedule('assign')
+    array_type = ArrayType(INTEGER_TYPE, [10, 10, 10])
+    fake_parent.symbol_table.add(DataSymbol('x', array_type))
+    fake_parent.symbol_table.add(DataSymbol('i', INTEGER_TYPE))
+    fake_parent.symbol_table.add(DataSymbol('j', INTEGER_TYPE))
+    fake_parent.symbol_table.add(DataSymbol('z', INTEGER_TYPE))
+
+    processor = Fparser2Reader()
+    processor.process_nodes(fake_parent, [fparser2part_ref])
+    # Check a new node was generated and connected to parent
+    assert len(fake_parent.children) == 1
+    new_node = fake_parent[0].lhs
+    assert isinstance(new_node, ArrayReference)
+    assert new_node.name == "x"
+    assert len(new_node.children) == 3  # Array dimensions
+
+
+@pytest.mark.usefixtures("f2008_parser")
+def test_handling_part_ref_function(parser):
+    ''' Test that fparser2 Part_Ref is converted to the expected PSyIR
+    tree structure when there is a function.
+    '''
+    # Note, the function will be a part_ref but will have a
+    # RoutineSymbol
+    code = (
+        "module test_mod\n"
+        "contains\n"
+        "  subroutine test_sub()\n"
+        "    integer :: a\n"
+        "    integer :: i\n"
+        "    a = test_func(i)\n"
+        "  end subroutine\n"
+        "  integer function test_func(i)\n"
+        "    integer :: i\n"
+        "    test_func = i\n"
+        "  end function test_func\n"
+        "end module\n")
+    reader = FortranStringReader(code)
+    ast = parser(reader)
+    processor = Fparser2Reader()
+    node = processor.generate_psyir(ast)
```

Comment:
Please could you change this test so that it doesn't use either of the parser fixtures and uses the `fortran_reader` instead.

Reply:
Oh yes, that was well out of date.